### PR TITLE
v1.4: Full coverage tests for `model` package

### PR DESCRIPTION
**Pull request by ZHTang29**

### CODE REVIEW

**Comment by SeeuSim** [comment]
Please run `./gradlew build` before committing

**Review by filbertphang — APPROVED** [review]
tests are comprehensive and well-written! test names may need to be changed to fit the coding standard though (see below). Notable offenders are all the `test_...` method names. Otherwise, LGTM 👍 

> Underscores may be used in test method names using the following three part format `featureUnderTest_testScenario_expectedBehavior()`
> e.g. `sortList_emptyList_exceptionThrown()` `getMember_memberNotFount_nullReturned`

> Third part or both second and third parts can be omitted depending on what's covered in the test. For example, the test method `sortList_emptyList()` will test `sortList()` method for all variations of the 'empty list' scenario and the test method `sortList()` will test the `sortList()` method for all scenarios.

File: src/test/java/seedu/recipe/model/recipe/exceptions/RecipeDurationInvalidArgumentLengthExceptionTest.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+public class RecipeDurationInvalidArgumentLengthExceptionTest {
+
+    @Test
+    public void validRecipeDurationInvalidArgumentLengthException() {
+        Exception exception = assertThrows(RecipeDurationInvalidArgumentLengthException.class, () ->
+                RecipeDuration.of("15"));
+        String expectedMessage = "An argument list of invalid length was passed for a Recipe Duration."
+                + "\nEnsure it is of the following format: '{number OR decimal} {duration}; "
+                + "\nExample: `1 minute`, `1.5 hours`";
```
Comment (SeeuSim) [inline src/test/java/seedu/recipe/model/recipe/exceptions/RecipeDurationInvalidArgumentLengthExceptionTest.java:18]: make this a `static final` constant at the start of the test class so its easier to modify

File: src/test/java/seedu/recipe/model/recipe/ingredient/IngredientBuilderTest.java [hunk 2]
```diff
@@ -32,7 +34,16 @@ public void constructor_invalidArgs() {
     }
 
     @Test
-    public void test_toString() {
+    public void overloadedConstructor() {
+        Ingredient polishedRice = Ingredient.of("polished rice");
+        polishedRice.setCommonName("rice");
+        assertEquals(COMMAND_STRING_SAMPLE, new IngredientBuilder(polishedRice,
+                new IngredientInformation(IngredientQuantity.of("300 g"), "1 scoop", new String[]{"washed"},
```
Comment (SeeuSim) [inline src/test/java/seedu/recipe/model/recipe/ingredient/IngredientBuilderTest.java:41]: newline tab should be 4 spaces, not 8

**Review by SeeuSim — APPROVED** [review]

**Pull request merged**